classes/Language.php

```diff
@@ -808,11 +808,30 @@ public static function loadLanguages()
 				LEFT JOIN `'._DB_PREFIX_.'lang_shop` ls ON (l.id_lang = ls.id_lang)';
 
         $result = Db::getInstance()->executeS($sql);
+
+        foreach ($result as $row) {
+            $idLang = (int) $row['id_lang'];
+
+            if (!isset(self::$_LANGUAGES[$idLang])) {
+                self::$_LANGUAGES[$idLang] = $row;
+            }
+            self::$_LANGUAGES[$idLang]['shops'][(int) $row['id_shop']] = true;
+        }
+    }
+
+    public static function loadLanguagesLegacy()
+    {
+        self::$_LANGUAGES = array();
+
+        $result = Db::getInstance()->executeS('SELECT * FROM `'._DB_PREFIX_.'lang`');
+
         foreach ($result as $row) {
-            if (!isset(self::$_LANGUAGES[(int) $row['id_lang']])) {
-                self::$_LANGUAGES[(int) $row['id_lang']] = $row;
+            $idLang = (int) $row['id_lang'];
```

Comment:
I just put 
```php
$idLang = (int) $row['id_lang'];
```
It create all difference but in reality, just added `loadLanguagesLegacy` and little refacto